models/tasks.js

```diff
@@ -701,6 +736,33 @@ const markUnDoneTasksOfArchivedUsersBacklog = async (users) => {
   }
 };
 
+/**
+ * Fetches all incomplete tasks for given user IDs.
+ *
+ * @param {string[]} userIds - The IDs of the users to fetch incomplete tasks for.
+ * @returns {Promise<firebase.firestore.QuerySnapshot>} - The query snapshot object.
+ * @throws {Error} - Throws an error if the database query fails.
+ */
+const fetchIncompleteTasksByUserIds = async (userIds) => {
+  const COMPLETED_STATUSES = [DONE, COMPLETED];
+
+  if (!userIds || userIds.length === 0) {
+    return [];
+  }
+  try {
+    const incompleteTasksQuery = await tasksModel.where("assigneeId", "in", userIds).get();
```

Comment:
Are we fetching incomplete tasks?

Reply:
Yes, this is only for abandoned users here, which is passed as param for the function.
In the next lines based on the status of the tasks it's filtered again.